
--CURDIR--/typeinference_for_args_negative.fz:42:11: error 1: Incompatible types when passing argument in a call
  say (a3 1.0         " world!")
----------^^^
Actual type for argument #1 'a' does not match expected type.
In call to          : 'typeinference_for_args_negative.a3'
expected formal type: 'String'
actual type found   : 'f64'
assignable to       : 'f64'
for value assigned  : '1.0'
To solve this, you could change the type of the target 'a' to 'f64' or convert the type of the assigned value to 'String'.


--CURDIR--/typeinference_for_args_negative.fz:43:14: error 2: Incompatible types when passing argument in a call
  say (a3 10.as_f64   " world!")
-------------^^^^^^
Actual type for argument #1 'a' does not match expected type.
In call to          : 'typeinference_for_args_negative.a3'
expected formal type: 'String'
actual type found   : 'f64'
assignable to       : 'f64'
for value assigned  : '10.as_f64'
To solve this, you could change the type of the target 'a' to 'f64' or convert the type of the assigned value to 'String'.


--CURDIR--/typeinference_for_args_negative.fz:44:11: error 3: Incompatible types when passing argument in a call
  say (a3 1.1         " world!")
----------^^^
Actual type for argument #1 'a' does not match expected type.
In call to          : 'typeinference_for_args_negative.a3'
expected formal type: 'String'
actual type found   : 'f64'
assignable to       : 'f64'
for value assigned  : '1.1'
To solve this, you could change the type of the target 'a' to 'f64' or convert the type of the assigned value to 'String'.


--CURDIR--/typeinference_for_args_negative.fz:47:33: error 4: Incompatible types when passing argument in a call
  say (a4 "1.0"       " world!".byte_length)
--------------------------------^^^^^^^^^^^
Actual type for argument #2 'b' does not match expected type.
In call to          : 'typeinference_for_args_negative.a4'
expected formal type: 'String'
actual type found   : 'i32'
assignable to       : 'i32'
for value assigned  : '" world!".byte_length'
To solve this, you could change the type of the target 'b' to 'i32' or convert the type of the assigned value to 'String'.


--CURDIR--/typeinference_for_args_negative.fz:48:23: error 5: Incompatible types when passing argument in a call
  say (a4 "10.as_f64" 3.14)
----------------------^^^^
Actual type for argument #2 'b' does not match expected type.
In call to          : 'typeinference_for_args_negative.a4'
expected formal type: 'String'
actual type found   : 'f64'
assignable to       : 'f64'
for value assigned  : '3.14'
To solve this, you could change the type of the target 'b' to 'f64' or convert the type of the assigned value to 'String'.


--CURDIR--/typeinference_for_args_negative.fz:49:32: error 6: Incompatible types when passing argument in a call
  say (a4 "1.1"       " world!"=" monde!")
-------------------------------^
Actual type for argument #2 'b' does not match expected type.
In call to          : 'typeinference_for_args_negative.a4'
expected formal type: 'String'
actual type found   : 'bool'
assignable to       : 'bool'
for value assigned  : '='
To solve this, you could change the type of the target 'b' to 'bool' or convert the type of the assigned value to 'String'.


--CURDIR--/typeinference_for_args_negative.fz:30:6: error 7: Type inference from actual arguments failed due to incompatible types of actual arguments
  a3(a, b) => a+b    # *** error ***, mixes String and f64 in arg `a`
-----^
For the formal argument 'typeinference_for_args_negative.a3.a' the following incompatible actual arguments where found for type inference:
actual is value of type 'String' at --CURDIR--/typeinference_for_args_negative.fz:41:11:
  say (a3 "hello"     " world!")
----------^^^^^^^
actuals are values of type 'f64' at --CURDIR--/typeinference_for_args_negative.fz:42:11:
  say (a3 1.0         " world!")
----------^^^
and at --CURDIR--/typeinference_for_args_negative.fz:43:14:
  say (a3 10.as_f64   " world!")
-------------^^^^^^
and at --CURDIR--/typeinference_for_args_negative.fz:44:11:
  say (a3 1.1         " world!")
----------^^^


--CURDIR--/typeinference_for_args_negative.fz:31:9: error 8: Type inference from actual arguments failed due to incompatible types of actual arguments
  a4(a, b) => a+b    # *** error ***, mixes String, i32, f64 and bool in arg `b`
--------^
For the formal argument 'typeinference_for_args_negative.a4.b' the following incompatible actual arguments where found for type inference:
actual is value of type 'String' at --CURDIR--/typeinference_for_args_negative.fz:46:23:
  say (a4 "hello"     " world!")
----------------------^^^^^^^^^
actual is value of type 'i32' at --CURDIR--/typeinference_for_args_negative.fz:47:33:
  say (a4 "1.0"       " world!".byte_length)
--------------------------------^^^^^^^^^^^
actual is value of type 'f64' at --CURDIR--/typeinference_for_args_negative.fz:48:23:
  say (a4 "10.as_f64" 3.14)
----------------------^^^^
actual is value of type 'bool' at --CURDIR--/typeinference_for_args_negative.fz:49:32:
  say (a4 "1.1"       " world!"=" monde!")
-------------------------------^


--CURDIR--/typeinference_for_args_negative.fz:32:6: error 9: Type inference from actual arguments failed since no actual call was found
  a5(a, b) => a+b    # *** error ***, never called
-----^
For the formal argument 'typeinference_for_args_negative.a5.a' the type can only be derived if there is a call to 'typeinference_for_args_negative.a5'.


--CURDIR--/typeinference_for_args_negative.fz:32:9: error 10: Type inference from actual arguments failed since no actual call was found
  a5(a, b) => a+b    # *** error ***, never called
--------^
For the formal argument 'typeinference_for_args_negative.a5.b' the type can only be derived if there is a call to 'typeinference_for_args_negative.a5'.

10 errors.
